noxfile.py

```diff
@@ -13,11 +15,54 @@
     nox.options.sessions = ["pre-commit", "unit"]
 
 
+def set_iree_state():
+    """
+    Check if IREE is enabled and set the environment variable accordingly.
+
+    Returns
+    -------
+    str
+        "ON" if IREE is enabled, "OFF" otherwise.
+
+    """
+    state = "ON" if os.getenv("PYBAMM_IDAKLU_EXPR_IREE", "OFF") == "ON" else "OFF"
+    if state == "ON":
+        if sys.platform == "win32":
+            warnings.warn(
+                (
+                    "IREE is not enabled on Windows yet. "
+                    "Setting PYBAMM_IDAKLU_EXPR_IREE=OFF."
+                ),
+                stacklevel=2,
```

Comment:
Hi @jsbrittain, I see that IREE is available as a wheel on x86_64 Windows as well now – based on https://pypi.org/project/iree-compiler/#files?

Or is this because the `IDAKLU_EXPR_IREE` files do not compile on Windows yet?

Reply:
IREE does officially support Windows. However, when trying to build for the platform I ran into compile errors in the IREE codebase. Specifically, these were due to the use of 'compound literals' which are not part of the C++ standard and are not supported by MSVC (C4576). Since IREE is based on the MLIR/LLVM codebase it is probably more compatible with clang++ (which does provide this support, but is not our chosen toolchain). So this is going to need some further consideration. @agriyakhetarpal @martinjrobins